python/cudf/cudf/tests/test_column.py

```diff
@@ -84,13 +84,12 @@ def test_column_series_multi_dim(data):
         cudf.core.column.as_column(data)
 
 
-@pytest.mark.parametrize("data", [["1.0", "2", -3], ["1", "0.11", 0.1]])
-def test_column_series_misc_input(data):
-    psr = pd.Series(data)
-    sr = cudf.Series(data)
-
-    assert_eq(psr.dtype, sr.dtype)
-    assert_eq(psr.astype("str"), sr)
+@pytest.mark.parametrize(
+    "data", [[1, "1.0", "2", -3], [np.nan, 0, "null", cp.nan]]
+)
+def test_column_mixed_dtype(data):
+    with pytest.raises(TypeError):
+        cudf.Series(data)
```

Comment:
Any chance we can make sure we're raising a more descriptive error that mixed dtypes have been passed in?

Reply:
We can't actually as depending on which value has been found first the pyarrow throws different error, so we might have to check the error message and throw it accordingly, which wouldn't be ideal and fool proof.

Reply:
Got it. Lets leave it for now and we can adjust if someone raises an issue in confusion.